### PR TITLE
`Exam Mode`: Fix exam checklist for test exams

### LINKED ISSUE #5505: Exam Mode (Test Exams): Fix UI Related to Test Exams
### Describe the bug

The items
- Register students for the exam
- Generate all student exams
- Prepare exercise start
are currently misleading for test exams.

Each shows (what is presumably only) a warning, such as
- ":x: At least one student registered"
- ":x: All exams generated for registered students"

Additionally, the corresponding page for registering students shows the hint ":information_source: Students are automatically registered for the Test Exam when starting a Test Exam", but the UI for registering students is still shown (but of course causes cryptic errors if used).

(also a nitpick: "Test Exams" should probably not be capitalized if it's not part of a title, since it's not a proper noun)

### To Reproduce

Make a mock exam and go to the exam overview page, go through the things described above.

### Expected behavior

For test exams, always show that it is intentional that no students are registered/no student exams are created/no exercises are prepared, e.g. with the hint ":heavy_check_mark: Students are automatically registered for the test exam when starting a test exam."

### Screenshots

_No response_

### What browsers are you seeing the problem on?

Other (specify in "Additional context")

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by mallgaier**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Closes #5505

### Description
The exam checklist for test exams had some bugs, which are resolved with this PR:
- Students do not need to be registered by an instructor, but this step was displayed in the checklist. Additionally, the button has been removed from the button-row in the top.
- Student exams do not need to be genereated by the instructor, but this step was displayed in the checklist
- The exercises do not need to be prepared by an instructor, but this step was displayed in the checklist
- Since several steps (registering students, generating student exams and preparing assignments) are automated compared to the normal exam and therefore do not need to be performed by the user, a corresponding note has been added. 

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Log in to Artemis
2. Navigate to Course Administration -> Exams
3. Create a test exam
4. Verify, that the steps *Student Registration, Generate Student Exams, Prepare Exercise Start* are not displayed
5. Verify, that the note regarding the special characteristics of the test exams can be displayed under *	Fill out all exam details*
6. Verify, that the button "Students" is no longer displayed
7. Create a real exam
8. Verify, that the checklist isn't broken

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

Bugfix

### CODE REVIEW

**Review by xlieA — APPROVED** [review]
Tested on ts1. Everything looks like described in the testing steps.

File: src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html [hunk 1]
```diff
@@ -115,92 +122,117 @@ <h3>
...
+                    <td>
+                        <span>{{ 'artemisApp.examManagement.checklist.tableItem.register' | artemisTranslate }}</span>
+                    </td>
+                    <td>
+                        <span>{{ 'artemisApp.examManagement.checklist.descriptionItem.register' | artemisTranslate }}</span>
+                        <hr />
+                        <ul class="list-unstyled">
+                            <li class="list-group-item border-1">
+                                <span>
+                                    <jhi-exam-checklist-check [checkAttribute]="!!exam.numberOfRegisteredUsers && exam.numberOfRegisteredUsers! > 0"></jhi-exam-checklist-check>
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html:137]:
```suggestion
                                    <jhi-exam-checklist-check [checkAttribute]="!!exam.numberOfRegisteredUsers"></jhi-exam-checklist-check>
```
This should be sufficient, or?

**Review by Dominik-Weinzierl — COMMENTED** [review]
Just a minor thing :+1:

**Review by bassner — APPROVED** [review]
Tested this according the steps, works as described. Have a look at the CR above, though :)

File: src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html (hunk 1 above)
Comment (krusche) [reply src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-checklist.component.html:137]: I think this was just taken over from the existing code. The changes here do not recognize that the code was there before and was just wrapped in an additional ngIf statement. I would just keep it and if at all change it in a follow up

**Comment by krusche** [comment]
Thanks for testing, I will merge now 👍

**Pull request merged**